### PR TITLE
fix(build-stats): esbuild and webpack stats under build-stats dir

### PR DESCRIPTION
**Pull request by JAdshead**
#### _Describe your changes in detail_

This relocates all webpack and esbuild stat files under a `.build-stats` directory. 

Addressing https://github.com/americanexpress/one-app-cli/issues/592

#### _Why is this change required? What issue does it resolve?_

See issue https://github.com/americanexpress/one-app-cli/issues/592

#### Check boxes that apply:
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (adding or updating documentation)
- [ ] Dependency update
- [ ] Security update

#### Check boxes that apply:
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] These changes should be applied to a maintenance branch.
- [ ] I have added the Apache 2.0 license header to any new files created.

### CODE REVIEW

**Comment by Matthew-Mallimo** [comment]
this could be considered a breaking change as many teams rely on these files for bundle analyzers

**Comment by JAdshead** [comment]
> this could be considered a breaking change as many teams rely on these files for bundle analyzers

i'm not sure that would be considered breaking. there is no change required to build the module and the assets created are dynamic. It does not impact any feature are alter requirements to run. Im happy to be wrong though

**Review by Matthew-Mallimo — DISMISSED** [review]

**Review by dogpatch626 — APPROVED** [review]

**Review by 10xLaCroixDrinker — APPROVED** [review]

File: packages/one-app-bundler/bin/postProcessOneAppBundle.js [hunk 1]
```diff
@@ -39,8 +39,11 @@ module.exports = async function postProcessBuild() {
   };
   const { hash } = await hashElement(tmpPath, options);
   const buildVersion = `${version}-${hash.slice(0, 8)}`;
-  const modernBrowserChunkAssets = require(path.resolve(pkgPath, '../.webpack-stats.browser.json')).assetsByChunkName;
-  const legacyBrowserChunkAssets = require(path.resolve(pkgPath, '../.webpack-stats.legacyBrowser.json')).assetsByChunkName;
+  if (!fs.existsSync(path.resolve(pkgPath, '../.build-stats'))) {
+    fs.mkdirSync(path.resolve(pkgPath, '../.build-stats'));
+  }
+  const modernBrowserChunkAssets = require(path.resolve(pkgPath, '../.build-stats/.webpack-stats.browser.json')).assetsByChunkName;
+  const legacyBrowserChunkAssets = require(path.resolve(pkgPath, '../.build-stats/.webpack-stats.legacyBrowser.json')).assetsByChunkName;
```
Comment (giulianok) [inline packages/one-app-bundler/bin/postProcessOneAppBundle.js:46]: solution could be simplified and make it a bit more readable

```suggestion
  const buildStatsPath = path.resolve(pkgPath, '../.build-stats');

  if (!fs.existsSync(buildStatsPath)) {
    fs.mkdirSync(buildStatsPath);
  }
  const modernBrowserChunkAssets = require(path.resolve(buildStatsPath, '.webpack-stats.browser.json')).assetsByChunkName;
  const legacyBrowserChunkAssets = require(path.resolve(buildStatsPath, '.webpack-stats.legacyBrowser.json')).assetsByChunkName;
```

File: packages/one-app-bundler/bin/webpackCallback.js [hunk 2]
```diff
@@ -42,7 +42,11 @@ module.exports = function getWebpackCallback(label, isModuleBuild) {
       });
     }
 
-    fs.writeFileSync(path.join(process.cwd(), `.webpack-stats.${label}.json`), JSON.stringify(jsonStats));
+    if (!fs.existsSync(path.join(process.cwd(), '.build-stats'))) {
```
Comment (giulianok) [inline packages/one-app-bundler/bin/webpackCallback.js:45]: same as https://github.com/americanexpress/one-app-cli/pull/595/files#r1424604318

**Comment by giulianok** [comment]
> > this could be considered a breaking change as many teams rely on these files for bundle analyzers
> 
> i'm not sure that would be considered breaking. there is no change required to build the module and the assets created are dynamic. It does not impact any feature are alter requirements to run. Im happy to be wrong though

the fact that `relocates all webpack and esbuild stat files under a '.build-stats' directory.` could be considered as a breaking change if, as @Matthew-Mallimo mentioned, teams rely on the current location. Do we know if teams actually rely on it?

**Comment by smackfu** [comment]
One other consideration is that currently these files are included in `.gitignore` so they are not checked in.  If moved to a new dir that is not git ignored, now people would see them show up as changed until they update their `.gitignore`.

Given that, the same issue came up with the move to use esbuild.

**Comment by JAdshead** [comment]
> One other consideration is that currently these files are included in `.gitignore` so they are not checked in. If moved to a new dir that is not git ignored, now people would see them show up as changed until they update their `.gitignore`.
> 
> Given that, the same issue came up with the move to use esbuild.

The filenames have not changed just the location so the `.gitignore` should still be valid. This will result in a empty directory which git should also ignore.

**Comment by JAdshead** [comment]
> > > this could be considered a breaking change as many teams rely on these files for bundle analyzers
> > 
> > 
> > i'm not sure that would be considered breaking. there is no change required to build the module and the assets created are dynamic. It does not impact any feature are alter requirements to run. Im happy to be wrong though
> 
> the fact that `relocates all webpack and esbuild stat files under a '.build-stats' directory.` could be considered as a breaking change if, as @Matthew-Mallimo mentioned, teams rely on the current location. Do we know if teams actually rely on it?

This does not affect the functionality or deployable build artifact so I don't consider it a breaking change. I can see it being considered an annoyance for anyone who has integrated bundle analysis into their pipelines.

**Comment by smackfu** [comment]
> The filenames have not changed just the location so the `.gitignore` should still be valid. This will result in a empty directory which git should also ignore.

Oh, that's awesome.  Just tested this and yep. Don't even need to ignore the .build-stats dir.

**Review by 10xLaCroixDrinker — APPROVED** [review]

**Review by Matthew-Mallimo — APPROVED** [review]

**Pull request merged**